### PR TITLE
Adds objective logging

### PR DESCRIPTION
**Pull request by ike709**
Requested by @Crossedfall 

Coded while half-asleep, untested. I don't guarantee that this catches everything nor do I guarantee that there's not duplicates.

### CODE REVIEW

File: code/__HELPERS/_logging.dm [hunk 1]
```diff
@@ -58,6 +58,10 @@
 	if (CONFIG_GET(flag/log_game))
 		WRITE_LOG(GLOB.world_game_log, "GAME: [text]")
 
+/proc/log_objective(whom, objective, admin_involved)
```
Comment (ghost) [inline code/__HELPERS/_logging.dm:61]: Is it just me or did you forget to actually add `[objective]` to the log below

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/__HELPERS/_logging.dm [hunk 2]
```diff
@@ -58,6 +58,10 @@
...
+	if (CONFIG_GET(flag/log_objective))
+		WRITE_LOG(GLOB.world_objective_log, "OBJ: [key_name(whom)] was assigned the following objective [admin_involved ? "by [key_name(admin_involved)]" : "automatically"]:")
```
Comment (ike709) [inline code/__HELPERS/_logging.dm]:
```suggestion
		WRITE_LOG(GLOB.world_objective_log, "OBJ: [key_name(whom)] was assigned the following objective [admin_involved ? "by [key_name(admin_involved)]" : "automatically"]: [objective]")
```

File: code/__HELPERS/_logging.dm (hunk 1 above)
Comment (ike709) [reply code/__HELPERS/_logging.dm:61]: Indeed. This is why I don't code while half-asleep.

**Comment by qwertyquerty** [comment]
If only you could just pass the objective holder upon objective creation, and log in one spot

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Missing the `GLOB.world_objective_log` entry in world.dm. This wont generate any logs as is

**Comment by ike709** [comment]
@Crossedfall Fixed.

**Comment by Crossedfall** [comment]
Test merged onto Golden

**Review by Crossedfall — APPROVED** [review]
Tests were positive. There may still be some antags that don't log properly, but as far as I've been able to see everything looks to be working as expected.

**Pull request merged**